requirements.txt

```diff
@@ -16,6 +16,6 @@ pyyaml >= 3.13, < 4.3
 pytz >= 2018.7
 requests >= 2.20, < 3.0
 toml >= 0.9.4, < 1.0
-typing >= 3.6.4, < 4.0
+typing >= 3.6.4, < 4.0; python_version < "3.5"
```

Comment:
Please correct me if I'm wrong, but doesn't this environment marker translate to: "only install `typing` if the system python version is < 3.5"?  Prefect doesn't support < 3.5 so I don't believe this will have any effect (as `typing` will still be installed as a dependency of `typing_extensions` below anyway).

Reply:
Fair point on 3.5 - however, if I'm interpreting correctly, it looks like `typing-extensions` only installs `typing` on Python < 3.5:

see: https://github.com/python/typing/blob/master/typing_extensions/setup.py#L55
```
...
if sys.version_info < (3, 5):
    install_requires.append('typing >= 3.6.2')
...
```

Reply:
so, does this mean we could just remove `typing` as a requirement entirely, then?

Reply:
I believe that may be correct, we could remove it entirely. The [package description](https://pypi.org/project/typing/) notes:

>NOTE: in Python 3.5 and later, the typing module lives in the stdlib, and installing this package has NO EFFECT.

Reply:
Very good point - at this exact moment, we could remove it; however, `typing` is included in the stdlib on a [_provisional_ basis](https://docs.python.org/3/glossary.html#term-provisional-api) with no guarantees of breaking changes.  Given that and given that I'm risk averse to these things, I'd prefer to keep version bounds in `requirements.txt`.

Reply:
Understood, and the notes in the package description suggest that something else is at the root of @joshbode's issue (as the requirement should have no effect)

Reply:
and from `typing`, the following imports are used:
```
Any
Callable
Dict
Generator
Iterable
Iterator
List
Mapping
MutableMapping
NamedTuple
Optional
Set
TYPE_CHECKING
Tuple
Union
cast
```
all of which seem to import fine in both 3.5 and 3.6 without `typing` or `typing-extensions` being installed

Reply:
fair enough on the provisionality of `typing` - I've found the source of the issue in the way the `black` vim plugin is loaded and made a PR there (basically it is inserting `site-packages` with a higher precedence than the system packages) so this PR is kind of unnecessary in that light

Reply:
Ok great!  Thanks for working through this with us, and glad to hear you were able to figure it out.  Given what we've learned, I'll close this PR but let us know if something comes up again in the future!

Reply:
will do - thanks for your help :)